src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
```

Comment:
[NAMING] - use proper names for object copy. Instead of `currentState`, a more descriptive name like `stateCopy` or `newState` would be better to indicate that this variable represents a modified copy of the original `state` object.